aten/src/ATen/native/sparse/SparseCsrTensor.cpp

```diff
@@ -157,17 +190,17 @@ SparseCsrTensor new_csr_tensor(const TensorOptions& options) {
   }
 
   return detail::make_tensor<SparseCsrTensorImpl>(
-      DispatchKeySet(dispatch_key), options.dtype());
+    DispatchKeySet(dispatch_key), options.layout(), options.dtype());
 }
 
 Tensor _sparse_csr_tensor_unsafe(const Tensor& crow_indices, const Tensor& col_indices,
     const Tensor& values,
     IntArrayRef size,
     c10::optional<ScalarType> dtype,
-    c10::optional<Layout> layout,
+    c10::optional<Layout> layout,  // unsafe requires explicit layout
```

Comment:
This could cause serialization errors when loading old models https://github.com/pytorch/pytorch/blob/30653d164d7223263e41c31060f75ad0bf9d9cab/torch/_utils.py#L187-L194
This call site isn't passed a layout.

Reply:
`layout` is now a part of the tensor data and it has the same level of importance as `crow_indices`, `col_indices`, `values`, and `size`. It is specified as optional for BC.

Notice that the content of `_rebuild_sparse_csr_tensor` now contains:
```python
result = torch._sparse_csr_tensor_unsafe(crow_indices, col_indices, values, size, layout=torch.sparse_csr)
```
that is, loading old models are ok.

However, we might have a problem with loading new models, let me check...

Reply:
Fixed.